run/opencl/argon2_kernel.cl

```diff
@@ -0,0 +1,424 @@
+//-------------------------------------------------------------------------------------
+// JtR OpenCL format to crack hashes from argon2.
+//
+// This software is copyright (c) 2023, Alain Espinosa <alainesp at gmail.com> and it
+// is hereby released to the general public under the following terms:
+//
+// Redistribution and use in source and binary forms, with or without
+// modification, are permitted.
+//
+//-------------------------------------------------------------------------------------
+//
+// Based on OpenCL code from https://gitlab.com/omos/argon2-gpu.
+// Copyright (c) 2016 Ondrej Mosnáček

```

Comment:
If what we have here is a derived work to an extent that @WOnder93 is a copyright holder, then we have to use his MIT license for this source file. That license says: "this permission notice shall be included in all copies or substantial portions of the Software." It says "_this_ permission notice" (emphasis mine), so we should not replace MIT's permission notice with ours.

Alternatively, we need @WOnder93's permission to use our 0-clause BSD for the derived work.

Reply:
Happy to see my 7 years old code being useful :)

I'd prefer to keep the licensing as is. Does having an MIT-licensed code in `john` pose a problem for you? If yes, I can consider dual-licensing it also under 0BSD.

Reply:
@WOnder93 This isn't really a problem, but if you simply reply here saying we can relicense the code under 0BSD, that would make things more consistent for us. There's no need for you to make changes in your repo. Thank you!

I think we'll be switching this code to use our shared header files e.g. providing device type detection and bit rotate macros, which are under 0BSD.